pkg/tsdb/grafana-pyroscope-datasource/service.go

```diff
@@ -55,50 +58,110 @@ func newInstanceSettings(httpClientProvider httpclient.Provider, ac accesscontro
 }
 
 func (s *Service) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	loggerWithContext := s.logger.FromContext(ctx)
+	loggerWithContext.Debug("Processing queries", "queries", req.Queries)
```

Comment:
Not certain here if there will be sensitive info in the logged query and someone without the proper rights will see the logged info. Would probably just remove logs with query to be safe.

Reply:
hmm not sure about that, I mean I wouldn't log the full request as to not log some authentication strings but I think logging a query can be super useful for debugging.